sensor/extrinsic_mapping_based_calibrator/src/base_lidar_calibrator.cpp

```diff
@@ -102,166 +78,90 @@ bool BaseLidarCalibrator::calibrate(
+    return std::make_tuple<>(false, Eigen::Matrix4d::Identity(), 0.f);
+  }
 
-    if (
-      inlier_size > parameters_->base_lidar_min_plane_points_ &&
-      inlier_percentage > parameters_->base_lidar_min_plane_points_percentage_ &&
-      cos_distance < parameters_->base_lidar_max_cos_distance_) {
-      RCLCPP_INFO(
-        rclcpp::get_logger(calibrator_name_), "Plane found: inliers=%ld (%.3f)",
-        inliers->indices.size(), inlier_percentage);
-      RCLCPP_INFO(
-        rclcpp::get_logger(calibrator_name_), "Plane model. a=%.3f, b=%.3f, c=%.3f, d=%.3f",
-        model(0), model(1), model(2), model(3));
-      RCLCPP_INFO(
-        rclcpp::get_logger(calibrator_name_), "Cos distance: %.3f / %.3f", cos_distance,
-        parameters_->base_lidar_max_cos_distance_);
+  publishResults(
+    ground_plane_model, last_keyframe->pose_, ground_plane_inliers_ptr, augmented_pointcloud_ptr);
 
-      // Extract the ground plane inliers
-      extract.setInputCloud(iteration_cloud);
-      extract.setIndices(inliers);
-      extract.setNegative(false);
-      extract.filter(*inliers_pointcloud);
-      return true;
-    }
+  Eigen::Isometry3d calibrated_base_to_lidar_transform =
+    tier4_ground_plane_utils::estimateBaseLidarTransform(
+      initial_base_to_lidar_transform, ground_plane_model);
 
-    // Extract the inliers from the pointcloud (the detected plane was not the ground plane)
-    extract.setInputCloud(iteration_cloud);
-    extract.setIndices(inliers);
-    extract.setNegative(true);
+  if (parameters_->base_lidar_overwrite_xy_yaw_) {
+    geometry_msgs::msg::TransformStamped initial_base_to_lidar_transform_msg_ =
+      tf2::eigenToTransform(initial_base_to_lidar_transform);
+    geometry_msgs::msg::TransformStamped calibrated_base_to_lidar_transform_msg =
+      tf2::eigenToTransform(calibrated_base_to_lidar_transform);
 
-    pcl::PointCloud<PointType> next_cloud;
-    extract.filter(next_cloud);
+    calibrated_base_to_lidar_transform_msg = tier4_ground_plane_utils::overwriteXYYawValues(
+      initial_base_to_lidar_transform_msg_, calibrated_base_to_lidar_transform_msg);
 
-    iteration_cloud->swap(next_cloud);
-    iteration_size = iteration_cloud->height * iteration_cloud->width;
+    calibrated_base_to_lidar_transform =
+      tf2::transformToEigen(calibrated_base_to_lidar_transform_msg);
   }
 
-  return false;
+  // Other calibrators look for lidar -> calibration_frame, so we follow suit
+  return std::make_tuple<>(true, calibrated_base_to_lidar_transform.inverse().matrix(), 0.f);
 }
 
 void BaseLidarCalibrator::publishResults(
-  const Eigen::Vector4d & ground_model,
-  const pcl::PointCloud<PointType>::Ptr & ground_plane_inliers_ptr,
-  const pcl::PointCloud<PointType>::Ptr & augmented_pointcloud_ptr)
+  const Eigen::Vector4d & ground_model, const Eigen::Matrix4f & pose,
+  const pcl::PointCloud<PointType>::Ptr & ground_plane_inliers_lcs_ptr,
```

Comment:
@.vividf
This comment was made on January 11, but unless you either start the review process (which I asked you not to) or mention me, I may not receive a notification.
The comment itself is outdated, since there was coding afterwards, making it difficult to address it

Reply:
Added the explanations in fc7fb7e4303ba9d679416271f74de5066e5fc5eb